### ISSUE TITLE
 Wrong number of args (3) passed to malli.core/-fail!

### ISSUE THREAD

**Issue by lucywang000** [metosin/malli#536]
The `-fail!` function accepts 1 or 2 args, but here apparently it gets 3 (2 plus one inserted by the `cond->` macro)

```
------ WARNING #1 - :fn-arity --------------------------------------------------
 Resource: malli/core.cljc:2007:43
--------------------------------------------------------------------------------
2004 |    (function-schema ?schema nil))
2005 |   ([?schema options]
2006 |    (let [s (schema ?schema options), t (type s)]
2007 |      (cond-> s (not (#{:=> :function} t)) (-fail! :invalid-=>schema {:type t, :schema s})))))
-------------------------------------------------^------------------------------
 Wrong number of args (3) passed to malli.core/-fail!
--------------------------------------------------------------------------------
```

**Comment by ikitommi** [comment]
This is fixed in master

**Comment by lucywang000** [comment]
Ah you're right. Sorry for the noise

**Comment by ikitommi** [comment]
noprob.

**Comment by solatis** [comment]
Could this potentially be cherry-picked and released as 0.6.2? It seems like this code path is currently broken in 0.6.1, and 0.7.0 is not yet released.
